### PR TITLE
* Add characterset option to MySQL database connection

### PR DESCRIPTION
**Pull request by flopraden**
Hi,

I loaded the nationalchar for utf-8 and utf-8 sqlauth isn't working since the MySQL connection wasn't configured for utf-8

Here the patch to add an option for character-set.
It may be interessant to put it for others SQL drivers

Best

### CODE REVIEW

File: src/modules/extra/m_mysql.cpp [hunk 1]
```diff
@@ -255,6 +255,15 @@ class SQLConnection : public SQLProvider
 		bool rv = mysql_real_connect(connection, host.c_str(), user.c_str(), pass.c_str(), dbname.c_str(), port, NULL, 0);
 		if (!rv)
 			return rv;
+		// enable character-set settings
+		std::string characterset;
+		if (config->readString("characterset", characterset))
+		{
+		  if (!mysql_set_character_set(connection,characterset.c_str()))
```
Comment (SaberUK) [inline src/modules/extra/m_mysql.cpp]: Your indentation is a bit wonky here. Tabs only please.

Also, you might want to fix the lack of a space after `connection,`

**Comment by flopraden** [comment]
It should be corrected.

File: src/modules/extra/m_mysql.cpp [hunk 2]
```diff
@@ -255,6 +255,15 @@ class SQLConnection : public SQLProvider
 		bool rv = mysql_real_connect(connection, host.c_str(), user.c_str(), pass.c_str(), dbname.c_str(), port, NULL, 0);
 		if (!rv)
 			return rv;
+		// enable character-set settings
+		std::string characterset;
+		if (config->readString("characterset", characterset))
+		{
+			if (!mysql_set_character_set(connection, characterset.c_str()))
```
Comment (attilamolnar) [inline src/modules/extra/m_mysql.cpp]: This check is incorrect, quote from the [manual](http://mirror.metrocast.net/mysql/doc/refman/5.5/en/mysql-set-character-set.html):

```
Return Values

Zero for success. Nonzero if an error occurred.
```

File: src/modules/extra/m_mysql.cpp [hunk 3]
```diff
@@ -255,6 +255,15 @@ class SQLConnection : public SQLProvider
 		bool rv = mysql_real_connect(connection, host.c_str(), user.c_str(), pass.c_str(), dbname.c_str(), port, NULL, 0);
 		if (!rv)
 			return rv;
+		// enable character-set settings
+		std::string characterset;
+		if (config->readString("characterset", characterset))
```
Comment (attilamolnar) [inline src/modules/extra/m_mysql.cpp]: Are you deliberately using `readString()` to allow empty strings? If not, please use `getString()` as the code above and test for empty() to determine whether or not to call the mysql_ func.

**Comment by flopraden** [comment]
In fact no.
I just use the same config-read block that m_mysql use for initialquery.
In fact, for initialquery, it should be rewrite this getString no?

**Comment by attilamolnar** [comment]
Thanks, squashed your commits and made some style changes to match our coding style.

**Comment by attilamolnar** [comment]
https://github.com/inspircd/inspircd/commit/b3bafc6aaee4b6b95c6aaefcc6f2e80eca6bd370

**Pull request closed without merging**